Make ModulesSelect remain open after selection

nexolute opened the pull request:
This PR is made with respect to #598.

I had implemented @yangshun original solution:

> One solution I have is to retain the search term and results screen so that the Added label appears and the user will know that the operation was successful. I think it's ok to leave the search term and results there as people usually add modules from the same faculty anyway.

ZhangYiJiang commented on www/src/js/views/timetable/ModulesSelect.jsx:
```diff
@@ -33,12 +33,25 @@ class ModulesSelect extends Component<Props, State> {
+  };
+
+  onStateChange = (changes: any) => {
+    if (Object.prototype.hasOwnProperty.call(changes, 'inputValue')) {
```
Just use `_.has` for this. Also, combine the two `if` statements

ZhangYiJiang commented on www/src/js/views/timetable/ModulesSelect.jsx:
```diff
@@ -33,12 +33,25 @@ class ModulesSelect extends Component<Props, State> {
+    if (Object.prototype.hasOwnProperty.call(changes, 'inputValue')) {
+      if (!Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
+        this.setState({
+          inputValue: changes.inputValue,
```
I see two bugs (not entirely sure if they are caused by this specific line, will investigate a bit further when I have time) 

- When blurring after selecting an item, this causes the input value to be set to the last select item's value (ie. it's module code), not the last input value 
- Cursor position is reset if you try to edit the middle of the search query

ZhangYiJiang commented:
Oh, and ignore the CircleCI failure - that's because of a config error on our part I think

nexolute commented:
Here's what I changed in the latest commit 4e95fac:

- Turns out the cursor issue is a [known issue](https://github.com/paypal/downshift/issues/217). I had fixed it accordingly.
- According to downshift documentation, props of rendered element should be passed as arguments into the [prop getter](https://github.com/paypal/downshift#prop-getters). I had changed it accordingly.
- I refactored the class.

ZhangYiJiang commented on www/src/js/views/timetable/ModulesSelect.jsx:
```diff
@@ -22,37 +22,63 @@ type Props = {
   isOpen: boolean,
+  isModalOpen: boolean,
+  inputValue: string,
+  selectedItem: any,
```
This should probably match the type of `item` property that was passed into `getItemProp`, which is `ModuleCode`. Since it is nullable, this should be `?ModuleCode`

yangshun commented:
@nexolute Good stuff, thanks for the PR!

ZhangYiJiang commented:
@nexolute Thanks for this PR, and sorry it took so long!
